File: KoKoslan/src/java/ast/KoKoDIV.java
/** 
   Andrey Arguedas Espinoza
   Daniela Armas Sanchez
   Michael Chen Wang
   Kimberly Olivas Delgado
*/

package kokoslan.java.ast;
import java.util.*;
import java.io.*;


public class KoKoDIV extends KoKoBiOperation {
	
    public KoKoDIV(KoKoAst operator, KoKoAst left, KoKoAst right){
		super(operator, left, right);
	}
	
	@Override
	public KoKoValue eval(KoKoContext ctx){
	   try {
		     KoKoNumValue lv = (KoKoNumValue)(left().eval(ctx));
			 KoKoNumValue rv = (KoKoNumValue)(right().eval(ctx));
	         return new KoKoNumValue(lv.getValue() / rv.getValue());
	   } catch (Exception e) {
			throw new KoKoEvalException(e.getMessage());
	   }
	   
   }
	
}

File: KoKoslan/src/kotlin/ast/KoKoDiv.kt
/** 
   Andrey Arguedas Espinoza
   Daniela Armas Sanchez
   Michael Chen Wang
   Kimberly Olivas Delgado
*/

package kokoslan.ast
import java.util.*
import java.io.*

class KoKoDIV(oper : KoKoAst, left : KoKoAst, right : KoKoAst) : KoKoBiOperation(oper, left, right) {
	
	override fun eval(ctx : KoKoContext) : KoKoValue{
	   try {
		     val lv = left().eval(ctx) as KoKoNumValue
			 val rv = right().eval(ctx) as KoKoNumValue
	         return KoKoNumValue(lv.getValue() / rv.getValue())
	   } catch (e : Exception) {
			throw KoKoEvalException(e.getMessage())
	   }   
   }
	
}